packages/toolpad-core/src/SignInPage/SignInPage.tsx

```diff
@@ -498,6 +519,89 @@ function SignInPage(props: SignInPageProps) {
               </Box>
             </React.Fragment>
           ) : null}
+
+          {emailProvider ? (
+            <React.Fragment>
+              {singleProvider ? null : <Divider sx={{ mt: 2, mx: 0, mb: 1 }}>or</Divider>}
+              {error && selectedProviderId === 'nodemailer' ? (
+                <Alert sx={{ my: 2 }} severity="error">
+                  {error}
+                </Alert>
+              ) : null}
+              {success && selectedProviderId === 'nodemailer' ? (
+                <Alert sx={{ my: 2 }} severity="success">
+                  {success}
+                </Alert>
+              ) : null}
+              <Box
+                component="form"
+                onSubmit={async (event) => {
+                  event.preventDefault();
+                  setFormStatus({
+                    error: '',
+                    selectedProviderId: emailProvider.id,
+                    loading: true,
+                  });
+                  const formData = new FormData(event.currentTarget);
+                  const emailResponse = await signIn?.(emailProvider, formData, callbackUrl);
+                  setFormStatus((prev) => ({
+                    ...prev,
+                    loading: false,
+                    error: emailResponse?.error,
+                    success: emailResponse?.success,
+                  }));
+                }}
+              >
+                <TextField
+                  margin="dense"
+                  required
+                  fullWidth
+                  slotProps={{
+                    htmlInput: {
+                      sx: { paddingTop: '12px', paddingBottom: '12px' },
```

Comment:
Maybe try to use theme units if possible unless we really need the 12px specifically here?